internal/parser.go

```diff
@@ -2499,6 +2499,9 @@ func frontmatterIM(p *parser) bool {
 }
 
 func inExpressionIM(p *parser) bool {
+	// p.afe (active formatting elements) should be cleared when entering an expression
+	p.clearActiveFormattingElements()
```

Comment:
This is the fix! When we encounter an expression, we should clear the context of `p.afe` (active formatting elements) to avoid mutating the authored content.